handler/forwarder/handler.go

```diff
@@ -105,65 +107,91 @@ func (h *Handler) WriteSQS(ctx context.Context, r io.Reader) error {
 	return nil
 }
 
-func (h *Handler) Handle(ctx context.Context, request events.SQSEvent) (response events.SQSEventResponse, err error) {
+func (h *Handler) ProcessRecord(ctx context.Context, record *events.SQSMessage) error {
 	logger := logr.FromContextOrDiscard(ctx)
 
-	var messages bytes.Buffer
-	defer func() {
-		if err == nil {
-			logger.V(3).Info("logging messages")
-			err = h.WriteSQS(ctx, &messages)
+	copyRecords := GetObjectCreated(record)
+	for _, copyRecord := range copyRecords {
+		sourceURL, err := url.Parse(copyRecord.URI)
+		if err != nil {
+			logger.Error(err, "error parsing source URI", "SourceURI", copyRecord.URI)
+			continue
 		}
-	}()
 
-	encoder := json.NewEncoder(&messages)
+		if !h.isBucketAllowed(sourceURL.Host) {
+			logger.Info("Received event from a bucket not in the allowed list; skipping", "bucket", sourceURL.Host)
```

Comment:
why logger.info here but logger.v(1).Info below?

Reply:
I just moved around the existing code, so not entirely sure. I assume that the rationale is that `V(0)` is used for a non-critical misconfiguration. You are subscribing a bucket, but did not grant the lambda permissions.

On the other hand, rejecting a file due to size is not a misconfiguration, it is the enforcement of a policy. There is no other way of preventing large files from triggering the lambda, since you can only subscribe based on prefix, and you may not control what is getting dumped into the bucket. As such it may warrant a higher verbosity.